library-best-practices/JLBP-9.md

```diff
@@ -0,0 +1,10 @@
+[JLBP-9] Ensure consumers always use a higher or equal minimum Java version
+---------------------------------------------------------------------------
+
+If a library B requiring Java 1.7 depends on a library A requiring Java 1.7, and
+library A bumps its minimum version to Java 1.8, then library B cannot pick up
+the new version of library A until it also bumps its own minimum Java
+version. This can result in a situation where the ecosystem is stuck on older
+library versions, preventing the pickup of important things like security
+fixes. Thus, libraries should make sure their consumers advance their minimum
+version of Java before they advance it themselves.
```

Comment:
This gives dependents veto power over advancement. In a world where many dependents are abandoned, I'm not sure I'm comfortable with that. 

I also think different teams move at different speeds, and I do like the idea of slowing down the advancement of minimum Java versions, so I really don't know what I think about this one.

Reply:
I would say that you can ignore abandoned dependents for the purposes of this rule; you have to wait for your consumers that "matter." I'll add a clarification.